web/src/app/components/header/header.component.ts

```diff
@@ -23,9 +23,23 @@ import {NavigationService} from 'app/services/navigation/navigation.service';
   styleUrls: ['./header.component.scss'],
 })
 export class HeaderComponent {
+  surveyId = '';
+  visible = false;
```

Comment:
This name, implies it controls the visibility of the whole header component rather than just the "survey settings" button.  But a more general question - why do we need to check the survey id at all? Can we just add an input and let the parent component decide when to show the button?

I'll approve for now since we're in a crunch, but if you agree with the above please add a code health issues and `TODO(#issueno)` to simplify this.